### DISCUSSION TITLE
Best practice for loading multiple humanoid models?

### DISCUSSION THREAD

**Discussion by jahaanjain** [pixiv/three-vrm#1089] [Help]
Hi there, working on a project where many humanoids will be rendered on the canvas, I noticed however that the frame rate drops significantly and starts lagging. Is there a good way I can render multiple humanoids (20+) at once with minimal (or none) lag?

**Comment by 0b5vr** [comment]
@jahaanjain 
We should start by finding where the bottleneck is.
Have you tried profiling, preferably from both CPU side and GPU side?

**Comment by beemsoft** [comment]
Hi, I did a bit of profiling and it seems the Spring bones calculation is taking quite some time for each model.
Did anyone already try to use Web Workers for the mathematical updates?
A similar approach has been done here:
https://github.com/EsProgram/uSpringBone

Kind regards,
Hans Beemsterboer

**Reply by 0b5vr** [reply]
SpringBone using JobSystem has also been introduced to the UniVRM recently, yes.
It seems we should consider optimizing the SpringBone system first.

**Comment by YulesRules** [comment]
@0b5vr Any progress on this? Otherwise are you able to suggest a way to gracefully disable spring joint/secondary animation calcs completely? : ]
